frame/staking-rewards/README.md

```diff
@@ -34,56 +33,74 @@ harvest `PICA` or `BTC` or `ETH` whenever a reward distribution occured.
+month duration with amount larger than existential deposit(ED). The reward
+multiplier is acting as a penalty on the computed share of the staking pool, the
+longer you stake, the higher your share. Let's say we have a reward multiplier
+of `0.8` for a `MONTH`, if I stake `X PICA`, my share will become `X * 0.8`.
+
+Once we staked `X` tokens, the pallet will mint a staking class `fNFT`
+representing our position. This NFT will hold the data required to compute our
+share and the reward we are able to claim at a time `t`. Like any other asset
+under Composable, this NFT is tradable. The NFT has an optional expiry date,
+which is the date at which it has been minted + the lock duration.
+
+### Unstaking while the NFT is locked (if lock is present)
 
 This case is called `early unstake` and will result in a penalty applied to the
 staked asset. Calling the `unstake` extrinsic on a NFT that is still locked will
 unlock it in exchange for a fraction of the stake (configured for the staked
 asset).
 
-Assuming the penalty is defined as being `0.5`, if I staked `10_000
-PICA`, unstaking the NFT will result in a penalty of  `5_000 PICA`. All
-harvested rewards are still in possession of the user.
+Assuming the penalty is defined as being `0.5`, if I staked `10_000 PICA`,
+unstaking the NFT will result in a penalty of `5_000 PICA`. All harvested
+rewards are still in possession of the user.
 
-### Unstaking once the NFT expired
+### Unstaking when the NFT has no lock duration or if the lock expired
 
 In this canonical case, the user is able to unstake and rapatriate it's staked
 asset with no penalty.
 
 ### Claiming
 
-A user is able to claim his pending rewards at any point in time.
-As the protocol is rewarding the NFT itself, all pending rewards are tied to the NFT only.
-Meaning that if a user trade a NFT including pending rewards, the new owner will be able to unlock them.
+A user is able to claim his pending rewards at any point in time. As the
+protocol is rewarding the NFT itself, all pending rewards are tied to the NFT
+only. Meaning that if a user trade a NFT including pending rewards, the new
+owner will be able to unlock them.
 
 ### Rewarding stakers
 
-Any protocol is able to rewards the stakers by calling the
-`StakingRewards::transfer_reward` implementation of this pallet.
+Any protocol is able to rewards specific asset stakers by calling
+`StakingRewards::transfer_reward`. The protocol must provide the `asset` to
+reward (a.k.a. fNFT minted for staking `asset`) as well as the `reward_asset`
+along the `reward` amount. Depending on the pallet state, the reward will either
+be stored in the `PendingEpochRewards` or `EpochRewards` storage.
+
+### Epochs
 
-### Epoches
+Rewards are accumulated in epochs.
 
-Rewards accumulated in epochs.
+A user is able to stake an amount to mint a fNFT, but the registration of the
+NFT to get rewarded will happen at the next epoch (i.e. a newly minted fNFT
+start getting rewarded at the next epoch only).
```

Comment:
```suggestion
An NFT will only start getting rewarded beginning with the next epoch (i.e. a newly minted fNFT
starts gathering rewards at the next epoch).
```
This could read like a new NFT (newly minted fNFT) is only rewarded once ever.
Maybe gathering isn't fitting in a technical sense though? Would appreciate some insight here.